### ISSUE TITLE
Can btree delete be more modular?

### ISSUE THREAD

**Issue by rkalyankumar** [mmnaseri/cs-review#6]
The private delete method runs over few hundres of lines of code. Can this be more modular?
https://github.com/mmnaseri/cs-review/blob/master/cs-clrs/src/main/java/com/mmnaseri/cs/clrs/ch18/s3/BTree.java#L31

**Comment by mmnaseri** [comment]
Good point. I'll put aside some time to do it this week, unless @rfarhanian wants to do it?

**Comment by rfarhanian** [comment]
I will do it.

**Comment by mmnaseri** [comment]
Thanks @rfarhanian for taking care of this :)
